compiler/stack_build.cmake

```diff
@@ -24,7 +24,7 @@ endif()
 set (buildGhcOptions "-O2")
 
 execute_process (
-    COMMAND ${STACK_EXECUTABLE} build :${target} --no-run-tests ${stack_options} --ghc-options=${buildGhcOptions}
+    COMMAND ${STACK_EXECUTABLE} build :${target} --no-run-tests ${stack_options} --ghc-options=${buildGhcOptions} ${BOND_STACK_OPTIONS}
```

Comment:
Neither `stack build` nor `stack path` take `--allow-different-user`, though neither seems to complain if `--allow-different-user` is passed to it.

Reply:
Both need it, because that is not a command-specific flag.